From dd9859f0976b3e447509ed664cb5261d77d063b9 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 22 Jul 2015 08:56:14 +0100
Subject: [PATCH 5/6] main.cpp: Remove duplicate _mesa_error_no_memory function

The stub implementation in src/glsl/standalone_scaffolding.cpp
was replaced by the implementation in src/glsl/main.cpp.
---
 src/glsl/main.cpp                   | 6 ------
 src/glsl/standalone_scaffolding.cpp | 2 ++
 2 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/glsl/main.cpp b/src/glsl/main.cpp
index feed100..97fcee1 100644
--- a/src/glsl/main.cpp
+++ b/src/glsl/main.cpp
@@ -40,12 +40,6 @@
 
 static int glsl_version = 330;
 
-extern "C" void
-_mesa_error_no_memory(const char *caller)
-{
-   fprintf(stderr, "Mesa error: out of memory in %s", caller);
-}
-
 static void
 initialize_context(struct gl_context *ctx, gl_api api)
 {
diff --git a/src/glsl/standalone_scaffolding.cpp b/src/glsl/standalone_scaffolding.cpp
index b338e92..16d4b47 100644
--- a/src/glsl/standalone_scaffolding.cpp
+++ b/src/glsl/standalone_scaffolding.cpp
@@ -32,6 +32,7 @@
 #include <assert.h>
 #include <string.h>
 #include <limits.h>
+#include <stdio.h>
 #include "util/ralloc.h"
 
 
@@ -52,6 +53,7 @@ _mesa_shader_debug(struct gl_context *, GLenum, GLuint *id,
 extern "C" void
 _mesa_error_no_memory(const char *caller)
 {
+    fprintf(stderr, "Mesa error: out of memory in %s", caller);
 }
 
 
-- 
2.4.5

